homeassistant/components/zha/sensor.py

```diff
@@ -56,279 +123,62 @@
-    if 'sub_component' in discovery_info:
-        sensor = discovery_info['sub_component'](**discovery_info)
-    elif RelativeHumidity.cluster_id in in_clusters:
-        sensor = RelativeHumiditySensor(**discovery_info)
-    elif PowerConfiguration.cluster_id in in_clusters:
-        sensor = GenericBatterySensor(**discovery_info)
-    elif TemperatureMeasurement.cluster_id in in_clusters:
-        sensor = TemperatureSensor(**discovery_info)
-    elif PressureMeasurement.cluster_id in in_clusters:
-        sensor = PressureSensor(**discovery_info)
-    elif IlluminanceMeasurement.cluster_id in in_clusters:
-        sensor = IlluminanceMeasurementSensor(**discovery_info)
-    elif Metering.cluster_id in in_clusters:
-        sensor = MeteringSensor(**discovery_info)
-    elif ElectricalMeasurement.cluster_id in in_clusters:
-        sensor = ElectricalMeasurementSensor(**discovery_info)
-        return sensor
-    else:
-        sensor = Sensor(**discovery_info)
-
-    return sensor
+    return Sensor(**discovery_info)
 
 
 class Sensor(ZhaEntity):
     """Base ZHA sensor."""
 
     _domain = DOMAIN
-    value_attribute = 0
-    min_report_interval = REPORT_CONFIG_MIN_INT
-    max_report_interval = REPORT_CONFIG_MAX_INT
-    min_reportable_change = REPORT_CONFIG_RPT_CHANGE
-    report_config = (min_report_interval, max_report_interval,
-                     min_reportable_change)
-
-    def __init__(self, **kwargs):
-        """Init ZHA Sensor instance."""
-        super().__init__(**kwargs)
-        self._cluster = list(kwargs['in_clusters'].values())[0]
-
-    @property
-    def zcl_reporting_config(self) -> dict:
-        """Return a dict of attribute reporting configuration."""
-        return {
-            self.cluster: {self.value_attribute: self.report_config}
-        }
-
-    @property
-    def cluster(self):
-        """Return Sensor's cluster."""
-        return self._cluster
 
-    @property
-    def state(self) -> str:
-        """Return the state of the entity."""
-        if isinstance(self._state, float):
-            return str(round(self._state, 2))
-        return self._state
-
-    def attribute_updated(self, attribute, value):
-        """Handle attribute update from device."""
-        _LOGGER.debug("Attribute updated: %s %s %s", self, attribute, value)
-        if attribute == self.value_attribute:
-            self._state = value
-            self.async_schedule_update_ha_state()
-
-    async def async_update(self):
-        """Retrieve latest state."""
-        result = await helpers.safe_read(
-            self.cluster,
-            [self.value_attribute],
-            allow_cache=False,
-            only_cache=(not self._initialized)
+    def __init__(self, unique_id, zha_device, listeners, **kwargs):
+        """Init this sensor."""
+        super().__init__(unique_id, zha_device, listeners, **kwargs)
+        sensor_type = kwargs.get(SENSOR_TYPE, GENERIC)
+        unit = UNIT_REGISTRY.get(sensor_type, None)
```

Comment:
`None` is the default value returned if the key is missing in the dict.

Reply:
it will sink in eventually :)